drivers/aks/aks_driver.go

```diff
@@ -1,74 +1,102 @@
 	"github.com/sirupsen/logrus"
 	"gopkg.in/yaml.v2"
+	"io/ioutil"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+	"regexp"
+	"strings"
+	"time"
 )
 
-var truePointer = true
 var redactionRegex = regexp.MustCompile("\"(clientId|secret)\": \"(.*)\"")
 
 type Driver struct {
 	driverCapabilities types.Capabilities
 }
 
 type state struct {
-	// Path to the public key to use for SSH into cluster
-	SSHPublicKeyPath string `json:"sshPublicKeyPath,omitempty"`
-
-	// Cluster Name
-	Name string
-
-	// The name that is displayed to the user on the Rancher UI
-	DisplayName string
-
+	/**
+	Azure Kubernetes Service API URI Parameters
+	*/
+	SubscriptionID string
+	ResourceGroup  string
+	Name           string
+
+	/**
+	Azure Kubernetes Service API Request Body
+	*/
+	/* Properties of Azure Active Directory configuration */
+	AzureADClientAppID     string
+	AzureADServerAppID     string
+	AzureADServerAppSecret string
+	AzureADTenantID        string
+
+	/* Properties for profile of managed cluster add-on, only support `httpApplicationRouting` and `monitoring` for now */
+	AddonHTTPApplicationRoutingEnabled  *bool
+	AddonMonitoringEnabled              *bool
+	LogAnalyticsWorkspaceResourceID     string
+	AddonHTTPApplicationRoutingDisabled *bool
+	AddonMonitoringDisabled             *bool
+
+	/* Properties of the agent pool. Currently only one agent pool can exist */
 	AgentDNSPrefix              string
+	AgentCount                  int64
+	AgentMaxPods                int64
+	AgentName                   string
+	AgentOsDiskSizeGB           int64
+	AgentStorageProfile         string
 	AgentVMSize                 string
-	Count                       int64
-	KubernetesVersion           string
-	Location                    string
-	OsDiskSizeGB                int64
-	SubscriptionID              string
-	ResourceGroup               string
-	AgentPoolName               string
-	MasterDNSPrefix             string
-	SSHPublicKeyContents        string
-	AdminUsername               string
-	BaseURL                     string
-	ClientID                    string
-	TenantID                    string
-	ClientSecret                string
+	VirtualNetworkResourceGroup string
 	VirtualNetwork              string
 	Subnet                      string
-	VirtualNetworkResourceGroup string
-	Tag                         map[string]string
-	ServiceCIDR                 string
-	DNSServiceIP                string
-	DockerBridgeCIDR            string
 
-	// Cluster info
+	/* Properties for profile of Linux VMs in the container service cluster. */
+	LinuxAdminUsername        string
+	LinuxSSHPublicKeyContents string
+	LinuxSSHPublicKeyPath     string
+
+	/* Properties for profile of Kubernetes network configuration */
+	NetworkDNSServiceIP     string
+	NetworkDockerBridgeCIDR string
+	NetworkPlugin           string
+	NetworkPolicy           string
+	NetworkPodCIDR          string
+	NetworkServiceCIDR      string
+
+	Location          string
```

Comment:
Should those parameters have prefix too? If there are new kinds of parameter, should you add comments for them?